### PR TITLE
First batch of new eslint rules

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Adds: "comma-style", "dot-location", "linebreak-style", "no-alert", "no-extra-bind", "no-lone-blocks", "no-nested-ternary", "no-undef-init", "no-useless-rename", "no-whitespace-before-property", "node/no-exports-assign", "rest-spread-spacing", "semi-spacing", "semi-style", "template-curly-spacing", "unicode-bom".

medic/cht-core#7440

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: config/default/test/forms/analytics/pregnancy.spec.js [hunk 1]
```diff
@@ -1,311 +1,311 @@
...
-        __patient_id: 'patient_id',
-        __household_uuid: 'patient_parent_id',
-        __source: 'action',
-        __source_id: ''
-      }
-    });
-  });
-
-});
+const { expect } = require('chai');
```
Comment (dianabarsan) [inline config/default/test/forms/analytics/pregnancy.spec.js:1]: This file had Windows line endings ...

Comment (m5r) [reply config/default/test/forms/analytics/pregnancy.spec.js:1]: Good job catching this one!

**Review by m5r — APPROVED** [review]
This gradual approach is the way to go, great job and thanks for taking care of our codebase!

Non-blocking question: when can we merge `7440-more-eslint` to `master`? I'm already dreading the eventual merge conflicts if we keep this branch alive for too long

**Comment by dianabarsan** [comment]
> when can we merge 7440-more-eslint to master

This will need to be AT-ed so I'm merging all eslint stuff into another branch (as per this PR) and then doing one massive merge and AT at the end. 
If things get out of control on number of conflicts, I can do an intermediary merge as well. 
This is to reduce the QA effort on this.

**Pull request merged**